### PR TITLE
[action] Update workflows

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  workflow_call:
+    secrets:
+      GITLEAKS_LICENSE_KEY:
+        required: true
+
+jobs:
+  ci:
+    uses: smallstep/workflows/.github/workflows/goCI.yml@main
+    with:
+      os-dependencies: "libpcsclite-dev"
```
Comment (maraino) [inline .github/workflows/ci.yml:19]: This is not a dependency of `step`

File: .github/workflows/code-scan-cron.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+on:
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (maraino) [inline .github/workflows/code-scan-cron.yml:3]: You're always using the same schedule, would we get any benefit if we use a different one per project? Perhaps a few minutes appart?

**Review by maraino — CHANGES_REQUESTED** [review]
The os-dependency is not necessary for this project.

File: .github/workflows/code-scan-cron.yml (hunk 2 above)
Comment (dopey) [reply .github/workflows/code-scan-cron.yml:3]: What would the benefit be? It's not running against our servers. Also, I changed every repo that isn't cli, certificates, crypto to only do once a week at midnight on sunday.

**Review by maraino — APPROVED** [review]

**Pull request merged**